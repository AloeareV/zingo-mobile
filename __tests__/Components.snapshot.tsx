/**
 * @format
 */

import 'react-native';
import React from 'react';

import { create } from 'react-test-renderer';
import {
  UsdAmount,
  ZecPrice,
  ZecAmount,
  BoldText,
  FadeText,
  ClickableText,
  ErrorText,
  RegText,
  RegTextInput,
} from '../components/Components';

jest.mock('react-native-localize', () => ({
  getNumberFormatSettings: () => {
    return {
      decimalSeparator: '.',
      groupingSeparator: ',',
    };
  },
}));
jest.useFakeTimers();

// test suite
describe('Component Components - test', () => {
  //snapshot test
  test('UsdAmount - snapshot', () => {
    const usdAmount = create(<UsdAmount price={null} amtZec={0} style={{}} />);
    expect(usdAmount.toJSON()).toMatchSnapshot();
  });

  test('ZecPrice - snapshot', () => {
    const zecPrice = create(<ZecPrice price={null} />);
    expect(zecPrice.toJSON()).toMatchSnapshot();
  });

  test('ZecAmount - snapshot', () => {
    const zecAmount = create(<ZecAmount color={''} size={0} amtZec={0} style={{}} zecSymbol={''} />);
    expect(zecAmount.toJSON()).toMatchSnapshot();
  });

  test('BoldText - snapshot', () => {
    const boldText = create(<BoldText style={{}} children={''} />);
    expect(boldText.toJSON()).toMatchSnapshot();
  });

  test('FadeText - snapshot', () => {
    const fadeText = create(<FadeText style={{}} children={''} />);
    expect(fadeText.toJSON()).toMatchSnapshot();
  });

  test('ClickableText - snapshot', () => {
    const clickableText = create(<ClickableText style={{}} children={''} />);
    expect(clickableText.toJSON()).toMatchSnapshot();
  });

  test('ErrorText - snapshot', () => {
    const errorText = create(<ErrorText style={{}} children={''} />);
    expect(errorText.toJSON()).toMatchSnapshot();
  });

  test('RegText - snapshot', () => {
    const regText = create(<RegText style={{}} color={''} onPress={() => {}} children={''} />);
    expect(regText.toJSON()).toMatchSnapshot();
  });

  test('RegTextInput - snapshot', () => {
    const regTextInput = create(<RegTextInput style={{}} />);
    expect(regTextInput.toJSON()).toMatchSnapshot();
  });
});
