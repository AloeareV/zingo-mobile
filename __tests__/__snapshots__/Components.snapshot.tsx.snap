// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Components - test BoldText - snapshot 1`] = `
<Text
  style={
    Array [
      Object {},
      Object {
        "color": "rgb(28, 28, 30)",
      },
      Object {
        "fontSize": 18,
      },
      Object {
        "fontWeight": "bold",
      },
      Object {
        "opacity": 0.87,
      },
    ]
  }
>
  
</Text>
`;

exports[`Component Components - test ClickableText - snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(28, 28, 30)",
        "textDecorationLine": "underline",
      }
    }
  >
    
  </Text>
</View>
`;

exports[`Component Components - test ErrorText - snapshot 1`] = `
<Text
  style={
    Object {
      "color": "red",
    }
  }
>
  
</Text>
`;

exports[`Component Components - test FadeText - snapshot 1`] = `
<Text
  style={
    Object {
      "color": "rgb(28, 28, 30)",
      "opacity": 0.65,
    }
  }
>
  
</Text>
`;

exports[`Component Components - test RegText - snapshot 1`] = `
<Text
  onPress={[Function]}
  style={
    Array [
      Object {},
      Object {
        "color": "rgb(28, 28, 30)",
        "fontSize": 18,
        "fontWeight": "600",
        "opacity": 0.87,
      },
    ]
  }
>
  
</Text>
`;

exports[`Component Components - test RegTextInput - snapshot 1`] = `
<TextInput
  allowFontScaling={true}
  editable={false}
  rejectResponderTermination={true}
  style={
    Array [
      Object {},
      Object {
        "color": "rgb(28, 28, 30)",
      },
      Object {
        "fontWeight": "600",
      },
    ]
  }
  underlineColorAndroid="transparent"
/>
`;

exports[`Component Components - test UsdAmount - snapshot 1`] = `
<Text
  style={
    Object {
      "color": "rgb(28, 28, 30)",
    }
  }
>
  $ 
  --
</Text>
`;

exports[`Component Components - test ZecAmount - snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "baseline",
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(28, 28, 30)",
        "fontSize": 24,
      }
    }
  >
    ZEC
     
    0.0000
  </Text>
  <Text
    style={
      Object {
        "color": "rgb(28, 28, 30)",
        "fontSize": 12,
        "paddingBottom": 0,
      }
    }
  >
    0000
  </Text>
</View>
`;

exports[`Component Components - test ZecPrice - snapshot 1`] = `
<Text
  style={
    Object {
      "color": "rgb(28, 28, 30)",
      "opacity": 0.65,
    }
  }
>
  
</Text>
`;
